tests/unit/query_helpers/test_question_validator.py

```diff
@@ -20,16 +23,35 @@ def test_is_query_helper_subclass():
 
 
 def test_passing_parameters():
-    """Test that llm_params is handled correctly and without runtime error."""
+    """Test that generic_llm_params is handled correctly and without runtime error."""
     # it is needed to initialize configuration in order to be able
     # to construct QuestionValidator instance
     config.init_config("tests/config/valid_config.yaml")
 
     question_validator = QuestionValidator()
-    assert question_validator.llm_params is not None
-    assert "max_new_tokens" in question_validator.llm_params
+    assert question_validator.generic_llm_params is not None
+    assert "max_new_tokens" in question_validator.generic_llm_params
+
+    question_validator = QuestionValidator(generic_llm_params={})
+    # the generic_llm_params should be rewritten in constructor
+    assert question_validator.generic_llm_params is not None
+    assert "max_new_tokens" in question_validator.generic_llm_params
+
+
+@patch("ols.src.query_helpers.question_validator.LLMChain", new=mock_llm_chain(None))
+def test_validate_question_llm_loader():
+    """Test that LLM is loaded within validate_question method with proper parameters."""
+    # it is needed to initialize configuration in order to be able
+    # to construct QuestionValidator instance
+    config.init_config("tests/config/valid_config.yaml")
+
+    llm_loader = mock_llm_loader(
+        None, expected_params=("p1", "m1", {"max_new_tokens": 4})
+    )
 
-    question_validator = QuestionValidator(llm_params={})
-    # the llm_params should be rewritten in constructor
-    assert question_validator.llm_params is not None
-    assert "max_new_tokens" in question_validator.llm_params
+    # check that LLM loader was called with expected parameters
```

Comment:
how is this actually verifying that it was called w/ the expected parameters?

Reply:
this is the trick:
```
llm_loader = mock_llm_loader(
        None, expected_params=("p1", "m1", {GenericLLMParameters.MAX_NEW_TOKENS: 4})
    )
```

I've added comment to make it a bit more clear